### PR TITLE
Clarifications and improvements to Kafka manual testing documentation

### LINKED ISSUE #322: [DOC] Kafka manual testing clarifications and improvements
## Report incorrect documentation
This was mentioned in PR #290 the documentation isn't completely clear on which lines from `CONTRIBUTING.md` need to be followed, and what commands are run on the Host OS, the Morpheus env/container or the Kafka container.

Along with the need to increase the file descriptors:
`ulimit -n 1048576` in the shell running morpheus/librdkafka

**Location of incorrect documentation**
`scripts/validation/kafka_testing.md`

### PR DESCRIPTION
**Pull request by dagardner-nv**
* Address feedback from #290 
* Add note about automated Kafka tests, which didn't exist at time the document was first written
* Fix spelling errors

Fixes #322

### CODE REVIEW

File: scripts/validation/kafka_testing.md [hunk 1]
```diff
@@ -362,26 +366,20 @@ For this test we are going to replace the from & to file stages from the Phishin
...
-1. Return to the Morpheus terminal. The pipeline will take anywhere from 2 to 5 minutes to complete. Once the `Kafka Write` monitor has reported that `1010` messages has been written shutdown Morpheus with Cntrl-C. We can check the number of lines in the outut file:
+1. Return to the Morpheus terminal. The pipeline will take anywhere from 2 to 5 minutes to complete. Once the `Kafka Write` monitor has reported that `1010` messages have been written, shutdown Morpheus with Ctrl-C. We can check the number of lines in the output file:
     ```bash
     wc -l ${MORPHEUS_ROOT}/.tmp/val_kafka_phishing.jsonlines
     ```
 
-1. Once all `1010` rows have been written, return to the Kafka terminal and stop the consumer with Cntrl-C.
+1. Once all `1010` rows have been written, return to the first Kafka terminal and stop the consumer with Ctrl-C.
 
 1. Verify the output with, expect to see `43` un-matched rows:
```
Comment (pdmack) [inline scripts/validation/kafka_testing.md:384]: Only got 19.
```
Results do not match. Diff 991/1010 (98.118812 %). First 10 mismatched rows:
Empty DataFrame
Columns: [data, pred]
Index: []```

**Review by pdmack — CHANGES_REQUESTED** [review]
@dagardner-nv @mdemoret-nv I wonder if the testing instructions should just stop at the basic copy, partition tests instead of the pipelines which are covered elsewhere. There's been a fair amount of divergence (e,g., HAMMAH->DFP) and I guess I'm worried that this will be a bit of a maintenance headache.

**Comment by dagardner-nv** [comment]
> I wonder if the testing instructions should just stop at the basic copy, partition tests instead of the pipelines which are covered elsewhere.

I think there is some truth to this, when I wrote the document I wanted to mimic each of the four pipelines in the `scripts/validation` directory. However the Kakfa version of those four pipelines have been codified into tests that are now run as a part of CI.

File: scripts/validation/kafka_testing.md [hunk 2]
```diff
@@ -1,26 +1,60 @@
...
+git clone https://gitlab.com/karolinepauls/pytest-kafka.git ${MORPHEUS_ROOT}/.cache/pytest-kafka
+cd ${MORPHEUS_ROOT}/.cache/pytest-kafka
+python setup.py develop
+cd ${MORPHEUS_ROOT}
+```
+
+Then run the Kafka tests with:
+```bash
+pytest --run_slow --run_kafka
+```
```
[the remark this answers is not in the document]
Comment (dagardner-nv) [reply scripts/validation/kafka_testing.md:14]: I think so, this appears to be errors when kafka & zookeeper are being shut down.

File: scripts/validation/kafka_testing.md [hunk 3]
```diff
@@ -1,26 +1,60 @@
+This document walks through manual testing of the Kafka functionality in Morpheus. There are also several automated tests which are run as part of the CI process. To run the tests locally first install `pytest-kafka` with:
+```bash
+mamba install -c conda-forge "openjdk=11.0.15"
```
[the remark this answers is not in the document]
Comment (dagardner-nv) [reply scripts/validation/kafka_testing.md:3]: > @dagardner-nv should this be changed to the apt install?

The advantage for the conda/mamba install is that it doesn't require root permissions.

**Comment by pdmack** [comment]
@gpucibot merge

**Pull request merged**